### ISSUE TITLE
Toolchain files should modify `CMAKE_SYSTEM_PREFIX_PATH`, not `CMAKE_PREFIX_PATH`.

### ISSUE THREAD

**Issue by jbadwaik** [microsoft/vcpkg#35469]
As seen [here](https://github.com/microsoft/vcpkg/blob/16ee2ecb31788c336ace8bb14c21801efb6836e4/scripts/buildsystems/vcpkg.cmake#L433) and [here](https://github.com/microsoft/vcpkg/blob/16ee2ecb31788c336ace8bb14c21801efb6836e4/scripts/buildsystems/vcpkg.cmake#L449), `vcpkg.cmake` modifies `CMAKE_PREFIX_PATH`. However, according to the [documentation](https://cmake.org/cmake/help/latest/variable/CMAKE_PREFIX_PATH.html), `CMAKE_PREFIX_PATH` belongs to the project and toolchains should instead set or modify `CMAKE_SYSTEM_PREFIX_PATH`.

**Comment by Neumann-A** [comment]
I think this boils done to personal preferences. I see vcpkg as part of the project and not as part of the system.

**Comment by jbadwaik** [comment]
I disagree, by project, the general meaning is the top-level `CMakeLists.txt` or a file included from that file. `vcpkg.cmake` is not used like that.
